src/command_modules/azure-cli-interactive/azclishell/telemetry.py

```diff
@@ -27,19 +27,53 @@ def my_context(tel_client):
     tel_client.context.instrumentation_key = INSTRUMENTATION_KEY
 
 
+class TelThread(threading.Thread):
+    """ telemetry thread for exiting """
+    def __init__(self, threadfunc, max_items=10):
+        threading.Thread.__init__(self)
+        self.threadfunc = threadfunc
+        self.counter = 0
+        self.max_items_until_flush = max_items
+
+    def force_run(self):
+        """ pushes the function out without a check """
+        try:
+            self.threadfunc()
+        except KeyboardInterrupt:
+            pass
+
+    def run(self):
+        try:
+            if self.counter >= self.max_items_until_flush:
```

Comment:
Also, couldn't that result in opening many more connections than necessary, which could be expensive?